### ISSUE TITLE
AccessRightParameterAssignment - TypeOfProofRef

### ISSUE THREAD

**Issue by janosu** [NeTEx-CEN/NeTEx#517]
Hi,

I needed to specify the TypeOfProofs in the UserProfile in addition to the predefined ProofOfIdentifyListOfEnum, so I updated the model to version 1.2.2. However, I noticed that in the case of fare products, I cannot refer to this in the AccessRightParameterAssignment validityParameters (only TypeOfConcessionRef is possible). I believe that, like TypeOfConcession, TypeOfProof is needed to indirectly define UserProfiles.

**Comment by Aurige** [comment]
You have an example (for new mode Car Club) in NeTEx\examples\functions\newModes\NewModes-CarClubExample.xml
I hope it can help
```
<GenericParameterAssignment order="1" id="my_car@membership@eligibility" version="any">
		<TypeOfAccessRightAssignmentRef versionRef="EXTERNAL" ref="fxc:eligible"/>
		<limitations>
			<!---  ===ELIGIBILITY USAGE PARAMETERS =========    -->
			<UserProfile version="any" id="adult_driver">
		         <Name>Qualified driver</Name>
			<UserType>adult</UserType>
			<MinimumAge>18</MinimumAge>
			**<typesOfProofRequiredRef>
				<TypeOfProofRef versionRef="external" ref="driving_licence"/>
			</typesOfProofRequiredRef>**
			</UserProfile>
		</limitations>
</GenericParameterAssignment>
```

**Comment by nick-knowles** [comment]
As Christophees exampel suggests, in principle a fare GENERIC PARAMETER ASSIGNMENT   would normally specify the eligible USER PROFILE, and the USER PROFILE would define what the TYPE(s) OF PROOF required are.   However it is possible to conceive circumstances where the type of proof might be a relevant  parameter assignment for a rule, for example if proof of identity must be presented alongside the ticket . 
So we should add TYPE OF PROOF as a parameter

UML for Transmodel
![image](https://github.com/NeTEx-CEN/NeTEx/assets/1826507/97065d91-fb04-4804-93e6-4ac23305e2e7)


UML For NeTEx
![image](https://github.com/NeTEx-CEN/NeTEx/assets/1826507/be44b84c-adda-47af-b634-bd0bf33141b8)

=========================
Also, iIn checking the UML I noticed   that the  documentation of  USER PROFILE needs changing to show the change to allow  USER PROFILEs to have multiple TYPEs OF PROOF

(1) Transmodel 
![image](https://github.com/NeTEx-CEN/NeTEx/assets/1826507/3c898aff-4ac0-42f3-9b99-158092d5c828)

**Comment by ue71603** [comment]
fixed in #599 
Perhaps still some TM adaption needed.
